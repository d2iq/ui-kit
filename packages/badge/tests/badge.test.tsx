import React from "react";
import { render } from "@testing-library/react";
import { Badge } from "../";

const StringComponent = (): JSX.Element => {
  return <span>string</span>;
};

describe("Badge", () => {
  it("default", () => {
    const { asFragment } = render(<Badge>default</Badge>);
    expect(asFragment()).toMatchSnapshot();
  });

  it("success", () => {
    const { asFragment } = render(<Badge appearance="success">success</Badge>);
    expect(asFragment()).toMatchSnapshot();
  });

  it("primary", () => {
    const { asFragment } = render(<Badge appearance="primary">primary</Badge>);

    expect(asFragment()).toMatchSnapshot();
  });

  it("danger", () => {
    const { asFragment } = render(<Badge appearance="danger">danger</Badge>);

    expect(asFragment()).toMatchSnapshot();
  });

  it("warning", () => {
    const { asFragment } = render(<Badge appearance="warning">warning</Badge>);

    expect(asFragment()).toMatchSnapshot();
  });

  it("outline", () => {
    const { asFragment } = render(<Badge appearance="outline">outline</Badge>);

    expect(asFragment()).toMatchSnapshot();
  });

  it("accept jsx as children", () => {
    const { asFragment } = render(
      <Badge>
        <StringComponent />
      </Badge>
    );

    expect(asFragment()).toMatchSnapshot();
  });
});
