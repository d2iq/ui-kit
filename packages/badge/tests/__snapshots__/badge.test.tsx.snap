// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Badge accept jsx as children 1`] = `
<span
  className="css-iu2at0"
  data-cy="badge"
>
  <span>
    string
  </span>
</span>
`;

exports[`Badge danger 1`] = `
<span
  className="css-1v2vcel"
  data-cy="badge"
>
  danger
</span>
`;

exports[`Badge default 1`] = `
<span
  className="css-iu2at0"
  data-cy="badge"
>
  default
</span>
`;

exports[`Badge outline 1`] = `
<span
  className="css-wrizyw"
  data-cy="badge"
>
  outline
</span>
`;

exports[`Badge primary 1`] = `
<span
  className="css-1bcogwf"
  data-cy="badge"
>
  primary
</span>
`;

exports[`Badge success 1`] = `
<span
  className="css-1nxr7gm"
  data-cy="badge"
>
  success
</span>
`;

exports[`Badge warning 1`] = `
<span
  className="css-dsjkko"
  data-cy="badge"
>
  warning
</span>
`;
